### DISCUSSION TITLE
Nuxt i18n dynamic slug generation

### DISCUSSION THREAD

**Discussion by luklus** [nuxt-modules/i18n#1291] [Q&A]
Hi, i have a little problem with dynamic slug.

i18n Config
```
i18n: {
    baseUrl: process.env.BASE_PAGE,
    defaultLocale: 'pl',
    detectBrowserLanguage: false,
    langDir: 'locales/',
    lazy: true,
    locales: [
      {
        code: 'en',
        iso: 'en-US',
        file: 'en-US.js',
      },
      {
        code: 'pl',
        iso: 'pl-PL',
        file: 'pl-PL.js',
      },
    ],
    seo: false,
  },
```

Nuxt Link
```
:to="
        localePath({
          path: `${$route.path}/${item.href}`,
        })
      "
```

What is the problem: 
Slug in Polish is for example: '/some-slug-in-polish'. But in English this slug is not exist, we have the same article in other, translated slug, for example: '/some-slug-in-english'. During generate I got the error:

```
Error generating route "/some-slug-in-english": Page not found 
Error generating route "/en/some-slug-in-polish": Page not found 
```

Did i miss something in the configuration?

**Comment by rchl** [comment]
Not enough information.
What's your `pages` file structure?
What's the value of `item.href` and where does it come from?

**Comment by luklus** [comment]
Page File Structure

/home
/_slug

Item href is come from CMS. As I mentioned, CMS is available in 2 languages: Polish and English. The English version is a Polish translation on the basis of:

``` "some-slug-in-english", "some-slug-in-polish"```

_slug page has an asyncData where:

``` const resp = await $prismic.api.getByUID(type, params.slug, { lang }) ```

**Comment by rchl** [comment]
Using `path` is not reliable. You should do something like: `localePath({ name: 'slug', params: { slug: item.href } })`

**Comment by luklus** [comment]
```:to="localePath({ name: 'slug', params: { slug: item.href } })"```

This one also did't work. Still the same error. The problem is that the English version generates a link for the English and Polish versions, and the Polish version generates a link for the Polish and English versions, not only for your language. 

item.href in PL is 'blog-post-pl' => generate 'blog-post-pl' and 'en/blog-post-pl' = 'en/blog-post-pl' is 404
item.href in En is 'blog-post-en' => generate 'blog-post-en' and 'en/blog-post-en' = 'blog-post-en' is 404

**Reply by rchl** [reply]
I guess you need to use "Dynamic route parameters" to specify which locale should support which slug. See https://i18n.nuxtjs.org/lang-switcher#dynamic-route-parameters

If this still doesn't help then please create a repo that reproduces since I don't have time to assemble and test the project myself.
